MetaMask/metamask-extension
# Capture exception with sentry when invariant conditions are met in migrations

## Explanation

There are states that we expect to never see in migrations. Although we do handle such cases and return state unchanged when they occur. This PR will add sentry `captureExceptions` calls when these states are detected, to better inform debugging of bugs/errors where the presence of those states may play a factor.

## Manual Testing Steps

1. Build this branch, onboard
2. Run the following script in the background console
```
chrome.storage.local.get(({ data, meta }) => chrome.storage.local.set({ data: { ...data, PreferencesController: false }, meta: {...meta, version: 81} }, () => { window.location.reload() }))
```
3. You should see a network request to sentry with  `typeof state.PreferencesController is boolean` in the payload

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
